### DISCUSSION TITLE
loading/error as default state in component store

### DISCUSSION THREAD

**Discussion by tomalaforge** [ngrx/platform#3154] [Ideas]
Hello ngrx team. 

Please find below, one small feature request which can be a nice addon to the component store. 

In componentStore, you often have to deal with loading indicator and error status, and you end up adding it in many of your componentState. 
Would it be nice to have it already set in the store, and automatically handle in the store effect function ? 

I have done a small stackblitz application to show you what I mean and how we can implement it. 
https://stackblitz.com/edit/angular-ivy-ggm1tl?file=src/app/component-store.ts

Waiting for your feedback

Thanks

**Comment by markostanimirovic** [comment]
Hello @tomalaforge  👋

`ComponentStore` is designed as a general solution for local state management. Having single loading/error status won't be a good fit in all cases. Adding specific things to a general-purpose state management library such as `component-store` can lead to limitations that we want to avoid.

Take a look at the [`DataComponentStore`](https://github.com/rx-mind/ngrx-plugins/tree/master/libs/data-component-store#readme). It has built-in support for CRUD operations (including pending statuses and error handling). Also, you can create a new plugin, if `DataComponentStore` doesn't suit your needs. 🙂

**Comment by markostanimirovic** [comment]
Moving this issue to Discussions (💡 Ideas category).

**Comment by tomalaforge** [comment]
@markostanimirovic Thank you for your answer. I thought so too but it was still beeing very generic compare to DataComponentStore.
